staging/src/k8s.io/component-base/metrics/registry.go

```diff
@@ -129,11 +144,11 @@ type kubeRegistry struct {
 // uniqueness criteria described in the documentation of metric.Desc.
 func (kr *kubeRegistry) Register(c Registerable) error {
 	if c.Create(&kr.version) {
+		defer kr.addResettable(c)
```

Comment:
Do you really want to add it to the resettable list if registering it gives an error?

Reply:
It's in a conditional, so it will only happen if Create returned true.

Reply:
Right, but the function could still return an error overall? (Honestly this function is already problematic, since it can return an error but leave things in a half completed state.)

Reply:
It shouldn't error in that spot, but we had to keep the signature consistent. But you are correct about this function being kinda broken, we should kick back an error if the create fails.